clash-prelude/src/Clash/Class/AutoReg/Internal.hs

```diff
@@ -113,6 +113,19 @@ class NFDataX a => AutoReg a where
   autoReg = register
   {-# INLINE autoReg #-}
 
+  -- | A "High-Speed" version of 'autoReg', that still splits product types
+  -- into multiple registers, but no longer uses the tag bit to control the
+  -- enable of the field bits. This reduces the potential logic depth, at the
+  -- cost of power due to toggling field bits.
+  autoRegHS
```

Comment:
Drive-by nitpick: `autoRegHs` maybe? Makes it easier to extend (`autoRegHsFoo` vs `autoRegHSFoo`) and a bit easier to type.

Reply:
Done